### DISCUSSION TITLE
Editing role permissions on the Permissions tab

### DISCUSSION THREAD

**Discussion by jonathanyeong** [supabase/supabase#720] [Questions] [answered]
Hey there! I'm a new Supabase user and I was wondering what is the best way to edit the `anon` role? I want to limit the role to Read Only.

From this page it looks like I should be able to check/uncheck roles but that's not the case.

<img width="1606" alt="Screen Shot 2021-02-14 at 6 55 23 pm" src="https://user-images.githubusercontent.com/3861088/107900572-3aa78780-6ef6-11eb-89d6-75faeb9c7210.png">

I looked at creating a new role, but it doesn't look like I can create a new API key that is assigned to that role. 

In the end, I ran `REVOKE INSERT on table from anon;` and other revoke commands in the SQL editor, and it worked! But I don't know if that's the best way to go about altering a role.

**Comment by silentworks** [comment, marked as answer]
@jonathanyeong I would suggest going another way about this as stated in the supabase guides by turning on row level security and setting up policies which would block none authenticated users from insert/update/delete on your tables. I think this approach is more flexible and allows for better customisation.

https://supabase.io/docs/guides/auth#row-level-security

**Reply by kiwicopple** [reply]
☝️ Definitely the way to go. I think we will turn off this screen for now because the Grants are more relavant to database-level access. There are few use-cases for this if you are developing an application - it's more relevant for attaching external tools

**Reply by jonathanyeong** [reply]
Okay perfect! Makes sense to me. I must have missed that step in the docs. Thanks @silentworks!
